pandas/tests/frame/test_query_eval.py

```diff
@@ -448,7 +448,7 @@ def test_date_index_query(self):
     def test_date_index_query_with_NaT(self):
         engine, parser = self.engine, self.parser
         n = 10
-        df = DataFrame(np.random.randn(n, 3))
+        df = DataFrame(np.random.randn(n, 3)).astype({0: object})
         df["dates1"] = date_range("1/1/2012", periods=n)
         df["dates3"] = date_range("1/1/2014", periods=n)
         df.iloc[0, 0] = pd.NaT
```

Comment:
here, `df[0]` is first created of `int` dtype, and then `df.iloc[0, 0] = pd.NaT` upcasts it to `object`. Might as well create it of dtype `object` in the first place, as the purpose of this test comes in the lines below (`df.query(...`)